### PR TITLE
🚑 Only create clients once in middleware

### PR DESCRIPTION
**Pull request by grdryn**
JIRA: https://issues.jboss.org/browse/FH-3355

### CODE REVIEW

**Comment by grdryn** [comment]
@wei-lee @paolobueno @aidenkeating Would one or more of you like to review?

I'll verify that this fixes the issue in the raincatcher-demo-cloud, once I've finished verifying that we don't run into the issue where it runs out of file descriptors without the timingMiddleware (in progress).

**Review by paolobueno — APPROVED** [review]
LGTM, next up would be running this on the demo-cloud setup for 'verification'

File: lib/timingMiddleware.js [hunk 1]
```diff
@@ -3,29 +3,29 @@ var types = require('./types');
 var MetricsClients = require('./clients');
 var _ = require('lodash');
 
-module.exports = function(component, metrics_conf) {
+module.exports = function timingMiddlewareCreator(component, metrics_conf) {
   var config = require('./config')(metrics_conf);
-  return function(req, res, next) {
+  var clients = new MetricsClients(config.getConfig());
```
Comment (wei-lee) [inline lib/timingMiddleware.js]: small thing but we only need to create the clients if it's enabled.

**Review by wei-lee — APPROVED** [review]

File: lib/timingMiddleware.js (hunk 1 above)
Comment (grdryn) [reply lib/timingMiddleware.js]: @wei-lee yeah, I guess I can put a check around this here too. 👍

**Comment by grdryn** [comment]
v2.5.1 published to npmjs, and tag pushed here.

**Pull request merged**